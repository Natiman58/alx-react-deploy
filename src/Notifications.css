.Notifications {
    border: 1px solid #ccc;
    padding: 20px;
    border: 1px solid red;
    border-style: dashed;
    margin: 20px;
    position: relative;
    font-family: 'Times New Roman', Times, serif;
}
.x-button {
    position: absolute;
    cursor: pointer;
    top: 10px;
    right: 10px;
    border: none;
    background: none;
}

li[data-priority="default"] {
    color: blue;
}

li[data-priority="urgent"] {
    color: red;
}