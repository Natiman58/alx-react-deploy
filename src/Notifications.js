import './Notifications.css';
import {getLatestNotification} from './utils'

function Notifications() {
    const handleClick = () => {
        console.log('Close button has been clicked');
    }
    return (
        <div className='Notifications'>
            <button className='x-button' aria-label="Close" onClick={handleClick}>x</button>
            <p>Here is the list of notifications</p>
            <ul>
                <li data-priority="default">New course available</li>
                <li data-priority="urgent">New resume available</li>
                <li data-priority="urgent" className="latest-notification" dangerouslySetInnerHTML={{ __html: getLatestNotification() }}></li>
            </ul>
        </div>
    )
}

export default Notifications