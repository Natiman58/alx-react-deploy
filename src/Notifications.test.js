import Notifications from './Notifications';
import renderer from 'react-test-renderer';

test('Notifications renders without crashing', () => {
    const component = renderer.create(<Notifications />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

test('Notifications renders 3 list items', () => {
    const component = renderer.create(<Notifications />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // To check that notifications renders 3 list items
    const notificationsList = tree.children[2].children.filter((child) => child.type === 'li');
    expect(notificationsList).toHaveLength(3);
});

test('Notifications reders the text "Here is the list of notifications" ', () => {
    const component = renderer.create(<Notifications />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // To check that notifications renders "Here is the list of notifications" text
    const paragraphText = tree.children[1].children[0];
    expect(paragraphText).toEqual("Here is the list of notifications");
});