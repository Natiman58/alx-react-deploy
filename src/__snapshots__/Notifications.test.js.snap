// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications reders the text "Here is the list of notifications"  1`] = `
<div
  className="Notifications"
>
  <button
    aria-label="Close"
    className="x-button"
    onClick={[Function]}
  >
    x
  </button>
  <p>
    Here is the list of notifications
  </p>
  <ul>
    <li
      data-priority="default"
    >
      New course available
    </li>
    <li
      data-priority="urgent"
    >
      New resume available
    </li>
    <li
      className="latest-notification"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<strong>Urgent requirement</strong> - complete by EOD",
        }
      }
      data-priority="urgent"
    />
  </ul>
</div>
`;

exports[`Notifications renders 3 list items 1`] = `
<div
  className="Notifications"
>
  <button
    aria-label="Close"
    className="x-button"
    onClick={[Function]}
  >
    x
  </button>
  <p>
    Here is the list of notifications
  </p>
  <ul>
    <li
      data-priority="default"
    >
      New course available
    </li>
    <li
      data-priority="urgent"
    >
      New resume available
    </li>
    <li
      className="latest-notification"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<strong>Urgent requirement</strong> - complete by EOD",
        }
      }
      data-priority="urgent"
    />
  </ul>
</div>
`;

exports[`Notifications renders without crashing 1`] = `
<div
  className="Notifications"
>
  <button
    aria-label="Close"
    className="x-button"
    onClick={[Function]}
  >
    x
  </button>
  <p>
    Here is the list of notifications
  </p>
  <ul>
    <li
      data-priority="default"
    >
      New course available
    </li>
    <li
      data-priority="urgent"
    >
      New resume available
    </li>
    <li
      className="latest-notification"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<strong>Urgent requirement</strong> - complete by EOD",
        }
      }
      data-priority="urgent"
    />
  </ul>
</div>
`;
